### PR TITLE
Remove Hosting.Internal APIs

### PR DESCRIPTION
**Pull request by Tratcher**
#11312 Everything became internal except IStartupConfigureContainerFilter and IStartupConfigureServicesFilter which became public but obsolete. These are only used in testing infrastructure, and only for WebHost.

I did need to clean up a mess where the Internal HostingApplication.Context had leaked into TestHost.

### CODE REVIEW

File: src/Hosting/TestHost/test/ClientHandlerTests.cs [hunk 1]
```diff
@@ -421,7 +420,7 @@ public Task ExceptionFromOnStartingWithErrorHandlerIsReported()
                 HttpCompletionOption.ResponseHeadersRead));
         }
 
-        private class DummyApplication : IHttpApplication<Context>
+        private class DummyApplication : ApplicationWrapper, IHttpApplication<HostingApplication.Context>
```
Comment (davidfowl) [inline src/Hosting/TestHost/test/ClientHandlerTests.cs]: This is the last thing to clean up or is this a test so it's fine?

Comment (Tratcher) [reply src/Hosting/TestHost/test/ClientHandlerTests.cs]: Just a test, I'm not worried about it.

Comment (davidfowl) [reply src/Hosting/TestHost/test/ClientHandlerTests.cs]: Sure, it's pretty horrible though

**Review by davidfowl — APPROVED** [review]

File: src/Hosting/TestHost/test/ClientHandlerTests.cs (hunk 1 above)
Comment (Tratcher) [reply src/Hosting/TestHost/test/ClientHandlerTests.cs]: Fixed

**Review by davidfowl — APPROVED** [review]

**Review by halter73 — APPROVED** [review]

**Comment by johnnyreilly** [comment]
When the `IStartupConfigureContainerFilter` is actually removed then this will create issues for Autofac users relying on this workaround: https://blog.johnnyreilly.com/2020/10/02/autofac-6-integration-tests-and-generic-hosting/

This is to do with `ConfigureTestContainer` not working with `GenericHost` - context here: https://github.com/dotnet/aspnetcore/issues/14907

**Pull request merged**